### PR TITLE
`clz` implementation and tests

### LINKED ISSUE #26: [Feature]: `clz` function
Implement function that count the number of leading zero bits in an integer.

Check out [Solidity Math Library](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/math/Math.sol) as a reference.

### PR DESCRIPTION
**Pull request by immrsd**
Resolves #26

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [ ] Changelog

## Release Notes

* **New Features**
  * Added count leading zeros (clz) support across all standard integer types (u8, u16, u32, u64, u128, u256) for efficient bit-level operations.

* **Tests**
  * Introduced comprehensive test suites for clz functionality, covering edge cases including zero values, maximum values, all bit positions, and boundary conditions.

### CODE REVIEW

**Review by ericnordelo — APPROVED** [review]
LGTM! Left a small non-blocking suggestion for consistency, but I think it is good to go otherwise.

Let's do some benchmarking after implementing log2 to see if the Solidity implementation is more efficient at least for u256 than the binary search.

NOTE: code can be formatted with `prettier-move "**/*.move" -w` if you have the prettier-move plugin installed.

File: math/core/tests/u128_tests.move [hunk 1]
```diff
@@ -130,3 +130,83 @@ fun mul_div_detects_overflow() {
...
 }
+
+// === clz ===
+
+// clz(0) should return 128 (all bits are leading zeros).
+#[test]
+fun clz_returns_bit_width_for_zero() {
+    let result = u128::clz(0);
+    assert_eq!(result, 128);
+}
```
[the remark this answers is not in the document]
Comment (immrsd) [reply math/core/tests/u128_tests.move]: Addressed

**Review by bidzyyys — APPROVED** [review]
Great job @immrsd!
One nit: Instead of `BIT_WIDTH` I would use `NUM_BITS` or just `BITS`.
